pkg/pillar/types/errortime.go

```diff
@@ -12,19 +12,95 @@ import (
+	RetryCountWarning = 10
+	//RetryTimeWarning indicates to move severity to Warning in case of more time spent for retry
+	RetryTimeWarning = time.Hour
+	//RetryCountError indicates to move severity to Error in case of more retries
+	RetryCountError = 20
+	//RetryTimeError indicates to move severity to Error in case of more time spent for retry
+	RetryTimeError = 2 * time.Hour
+)
+
+// ErrorSeverity tells the severity type, must be in sync with info.Severity enum of api
+type ErrorSeverity int32
+
+const (
+	// ErrorSeverityUnspecified severity unspecified
+	ErrorSeverityUnspecified ErrorSeverity = 0
+	// ErrorSeverityNotice severity notice
+	ErrorSeverityNotice ErrorSeverity = 1
+	// ErrorSeverityWarning severity warning
+	ErrorSeverityWarning ErrorSeverity = 2
+	// ErrorSeverityError severity error
+	ErrorSeverityError ErrorSeverity = 3
+)
+
+//GetErrorSeverity returns ErrorSeverity based on retry count and time spend
+func GetErrorSeverity(retryCount int, timeSpend time.Duration) ErrorSeverity {
+	if retryCount > RetryCountError || timeSpend > RetryTimeError {
+		return ErrorSeverityError
+	}
+	if retryCount > RetryCountWarning || timeSpend > RetryTimeWarning {
+		return ErrorSeverityWarning
+	}
+	return ErrorSeverityNotice
+}
+
+// ErrorEntityType contains the entity type, must be in sync with info.Entity enum of api
+type ErrorEntityType int32
+
+const (
+	// ErrorEntityUnspecified Entity
+	ErrorEntityUnspecified ErrorEntityType = 0
+	// ErrorEntityBaseOs entity
+	ErrorEntityBaseOs ErrorEntityType = 1
+	// ErrorEntitySystemAdapter Entity
+	ErrorEntitySystemAdapter ErrorEntityType = 2
+	// ErrorEntityVault Entity
+	ErrorEntityVault ErrorEntityType = 3
+	// ErrorEntityAttestation Entity
+	ErrorEntityAttestation ErrorEntityType = 4
+	// ErrorEntityAppInstance Entity
+	ErrorEntityAppInstance ErrorEntityType = 5
+	// ErrorEntityPort Entity
+	ErrorEntityPort ErrorEntityType = 6
+	// ErrorEntityNetwork Entity
+	ErrorEntityNetwork ErrorEntityType = 7
+	// ErrorEntityNetworkInstance Entity
+	ErrorEntityNetworkInstance ErrorEntityType = 8
+	// ErrorEntityContentTree Entity
+	ErrorEntityContentTree ErrorEntityType = 9
+	// ErrorEntityContentBlob Entity
+	ErrorEntityContentBlob ErrorEntityType = 10
+	// ErrorEntityVolume Entity
+	ErrorEntityVolume ErrorEntityType = 11
+)
+
+// ErrorEntity contains the device entity details
+type ErrorEntity struct {
+	EntityType ErrorEntityType // entity type
+	EntityID   string          // entity uuid, sha, or other unique id based on the type
+}
+
+// ErrorDescription contains error description to be passed into functions
+type ErrorDescription struct {
+	Error               string
+	ErrorTime           time.Time
+	ErrorSeverity       ErrorSeverity
+	ErrorRetryCondition string
+	ErrorEntities       []*ErrorEntity
+}
+
 // ErrorAndTime is used by many EVE agents
 type ErrorAndTime struct {
-	Error     string
-	ErrorTime time.Time
+	ErrorDescription
 }
 
 // SetErrorNow uses the current time
```

Comment:
I think these functions now become legacy/deprecated. Would make sense to state that new code should use SetErrorWithDescription()

Reply:
I put comments for this